platforms/macosx/src/GDeviceOSX.cpp

```diff
@@ -600,113 +586,17 @@ void GDeviceOSX::DrawMusicSymbol( float x, float y, unsigned int inSymbolID )
 			x -= (w * float(0.5));
 	}
 
-	::CGContextSelectFont(mContext, mCurrMusicFont->GetName(), mCurrMusicFont->GetSize(), kCGEncodingMacRoman  );// ok
+    // Uncommented code does not work. Core text uses different coordinates.
+    // CGPoint point = CGPointMake(x, y);
+    // CTFontDrawGlyphs(macFont->fCTFont, &glyph, &point, 1, mContext);
+    ::CGContextSetFont(mContext, macFont->fCGFont);
+    ::CGContextSetFontSize(mContext, macFont->GetSize());
 
 	// - Draw text
 	PushFillColor( VGColor(mTextColor.mRed, mTextColor.mGreen, mTextColor.mBlue, mTextColor.mAlpha) );
 	::CGContextShowGlyphsAtPoint(mContext, x, y, &glyph, 1 );
 	PopFillColor();
 }
-// --------------------------------------------------------------
-//#ifndef IOS
-#if 0
-void GDeviceOSX::DrawString( float x, float y, const char * s, int inCharCount )
-{
-	// this is for macos 10.4 : select a dummy font first 
-	::CGContextSelectFont(mContext, "Monaco", mCurrTextFont->GetSize(), kCGEncodingMacRoman  );// ok
-	
-	if (s == 0 || s[ 0 ] == '\0' || inCharCount == 0)
-		return;
-
-	if (inCharCount == -1) 
-		inCharCount = (int)strlen( s );
-
-	// - Manage character encoding
-	const char * convStr = s; 
-	if( convStr == 0 ) return;
-
-	// - Calculate string dimensions
-	float w = 0;
-	float h = 0;
-	float baseline = 0;
-	mCurrTextFont->GetExtent( convStr, inCharCount, &w, &h, this );
-		
-	// - Perform text alignement
-	if( mTextAlign != ( kAlignLeft | kAlignBase )) {
-		if( mTextAlign & kAlignBottom )	// Vertical align
-			y -= baseline; 
-		else if( mTextAlign & kAlignTop )
-			y += h - baseline;
-
-		if( mTextAlign & kAlignRight )	// Horizontal align
-			x -= w;
-		else if( mTextAlign & kAlignCenter )
-			x -= (w * 0.5);
-	}
-
-	// - Draw text background
-//	if( mTextBackColor.mAlpha < 255 ){
-		PushPen( mTextBackColor, 1 );
-		PushFillColor( mTextBackColor );
-		Rectangle( x, y, x + w, y + h );
-		PopFillColor();
-		PopPen();
-//	}
-
-	::CGContextSelectFont(mContext, mCurrTextFont->GetName(), mCurrTextFont->GetSize(), kCFStringEncodingMacRoman);
-    
-	// - Draw text
-	PushFillColor( VGColor(mTextColor.mRed, mTextColor.mGreen, mTextColor.mBlue,  mTextColor.mAlpha) );
-	::CGContextShowTextAtPoint(mContext, x, y, convStr, (size_t)inCharCount );
-	PopFillColor();
-	
-	// - Debug: draw refpoint
-//	PushPen( GColor( 250, 0, 0 ), 5 );	
-//	Line( debugX - 20, debugY + 20, debugX + 20, debugY - 20 );
-//	Line( debugX - 20, debugY - 20, debugX + 20, debugY + 20 );	
-//	PopPen(); 
-}
-#else		// defined IOS
```

Comment:
don't remove this code, it might be used by clients of the library

Reply:
there is a #else and from what I can see, this part is also removed (the fontName2iOSName function)

Reply:
It is not needed any more. It is replaced by the new Font initialization with FontDescriptor.